tests/components/solaredge/test_config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Tests for the SolarEdge config flow."""
+import pytest
+from unittest.mock import patch
+
+from homeassistant import data_entry_flow
+from homeassistant.components.solaredge import config_flow
+from homeassistant.components.solaredge.const import CONF_SITE_ID, DEFAULT_NAME
+from homeassistant.const import CONF_NAME, CONF_API_KEY
+
+from tests.common import MockConfigEntry
+
+NAME = "solaredge site 1 2 3"
+SITE_ID = "1a2b3c4d5e6f7g8h"
+API_KEY = "a1b2c3d4e5f6g7h8"
+
+
+@pytest.fixture(name="test_connect")
+def mock_controller():
+    """Mock a successfull _check_site."""
+    with patch(
+        "homeassistant.components.solaredge.config_flow.SolarEdgeConfigFlow._check_site",
```

Comment:
Don't patch our check site method. Patch the library api class and yield the mock. Then we can set different side effects on the mock in the different tests.

We should check all the different connection cases. Doing this will complete coverage.